gammapy/stats/counts_statistic.py

```diff
@@ -129,20 +138,26 @@ def n_sig_matching_significance(self, significance):
         n_sig : `numpy.ndarray`
             Excess
         """
+        from gammapy.estimators.utils import find_roots
+
         n_sig = np.zeros_like(self.n_bkg, dtype="float")
         it = np.nditer(n_sig, flags=["multi_index"])
 
         while not it.finished:
-            try:
-                # Note that this can fail silently now. A false root can be found. Because of too large slope?
-                n_sig[it.multi_index] = newton(
-                    self._n_sig_matching_significance_fcn,
-                    np.sqrt(self.n_bkg[it.multi_index]) * significance,
-                    args=(significance, it.multi_index),
-                )
-            except:
-                n_sig[it.multi_index] = np.nan
-
+            lower_bound = np.sqrt(self.n_bkg[it.multi_index]) * significance
+            # find upper bounds for secant method as in scipy
+            eps = 1e-4
+            upper_bound = lower_bound * (1 + eps)
```

Comment:
for the secant method two initial estimates x0 and x1 are needed, called from find_roots it will use the upper and lower bounds argument, but as upper bound cannot be None in find_roots  I set it as in  https://github.com/scipy/scipy/blob/4ec4ab8d6ccc1cdb34b84fdcb66fde2cc0210dbf/scipy/optimize/zeros.py#L317-L325